### DISCUSSION TITLE
creating-a-rolling-file-logging-provider-for-asp-net-core-2-0/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#227] [Announcements]
# creating-a-rolling-file-logging-provider-for-asp-net-core-2-0/

In this post I'll show how to create an ILoggingProvider for ASP.NET Core that writes logs to the file system, based on the Azure App Service provider.

https://andrewlock.net/creating-a-rolling-file-logging-provider-for-asp-net-core-2-0/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at November 22 2018, 01:06, in reply to [Benners](https://disqus.com/by/disqus_OEWMOfb2MT/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/227#discussioncomment-6042907)</em>

---

<p>Hey, glad you've found it useful. The short answer is that this logger doesn't currently support scopes. However, I did some work on it last night, and it should be possible soon, as long as you're using .NET Core 2.1.</p><p>Logging the method name is a separate issue in of itself, and is worthy of a blog post I think :) The approach shown in that stack overflow post is the way to do it, so you should be able to use that once I've added scope logging to the Rolling File provider.</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at November 22 2018, 05:23, in reply to [Ian Warburton](https://disqus.com/by/ian_warburton/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/227#discussioncomment-6042909)</em>

---

<p>Because its logging configuration, not middleware :) in 2.0+ logging configuration is part of the Webhostbuilder configuration, so it happens earlier in the bootstrapping process</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at January 09 2019, 06:32, in reply to [Rain Dropz](https://disqus.com/by/rain_dropz/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/227#discussioncomment-6042915)</em>

---

<p>As far as I'm aware there shouldn't be any problem using the library with 2.2. If you run into issues, feel free to raise an issue on GitHub 🙂 <a href="https://github.com/andrewlock/NetEscapades.Extensions.Logging/blob/master/src/NetEscapades.Extensions.Logging.RollingFile/" rel="nofollow noopener" title="https://github.com/andrewlock/NetEscapades.Extensions.Logging/blob/master/src/NetEscapades.Extensions.Logging.RollingFile/">https://github.com/andrewlock/NetEscapades.Extensions.Logging/blob/master/src/NetEscapades.Extensions.Logging.RollingFile/</a></p>

[a comment by an automated account is not shown]

**Comment by AndreduToit** [comment]
Andrew - every time I get stuck with something that I need to understand, somehow, your insights always seem to provide me with right and proper information and guidance. Like pondering "why use Serilog"? Well, I need to log to file and you said it "one provider is particularly conspicuous by its absence - a file provider". Thanks - for all the times.
